aries_cloudagent/revocation/indy.py

```diff
@@ -107,23 +110,60 @@ async def init_issuer_registry(
 
     async def handle_full_registry(self, revoc_reg_id: str):
         """Update the registry status and start the next registry generation."""
+        await self._set_registry_status_registry(
+            revoc_reg_id, IssuerRevRegRecord.STATE_FULL
+        )
+
+    async def decommission_registry(self, cred_def_id: str):
+        """Decommission post-init registries and start the next registry generation."""
+        async with self._profile.session() as session:
+            registries = await IssuerRevRegRecord.query_by_cred_def_id(
+                session, cred_def_id
+            )
+
+        # decommission everything except init
+        recs = list(
+            filter(lambda r: r.state != IssuerRevRegRecord.STATE_INIT, registries)
+        )
+
+        init = True
+        for rec in recs:
+            LOGGER.debug(f"decommission {rec.state} rev. reg.")
+            LOGGER.debug(f"revoc_reg_id: {rec.revoc_reg_id}")
+            LOGGER.debug(f"cred_def_id: {cred_def_id}")
+            await self._set_registry_status_registry(
+                rec.revoc_reg_id, IssuerRevRegRecord.STATE_DECOMMISSIONED, init
+            )
+            init = False  # only call init once.
+
+        return recs
+
+    async def _set_registry_status_registry(
+        self, revoc_reg_id: str, state: str, init: bool = True
+    ):
```

Comment:
It doesn't look like `init` is used anywhere inside this method? Also, a nitpick: I feel like there's one too many "registry" in the name of this method.

Reply:
haha, there certainly is one too many "registry" in that name. great catch on both accounts... i think the init got lost in a code merge/rebase.

touching up now.